src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -9145,7 +9145,7 @@ private PostSkipAction SkipBadForStatementExpressionListTokens(ref SyntaxToken s
 
             var openParen = this.EatToken(SyntaxKind.OpenParenToken);
 
-            var variable = ParseExpressionOrDeclaration(ParseTypeMode.Normal, feature: MessageID.IDS_FeatureTuples, permitTupleDesignation: true);
+            var variable = ParseExpressionOrDeclaration(ParseTypeMode.Normal, permitTupleDesignation: true);
```

Comment:
much simpler on the parser side.  This is another case where moving to binder simplified things.